edge-functions/api-rate-limit/middleware.ts

```diff
@@ -0,0 +1,14 @@
+import { type NextRequest, NextResponse } from 'next/server'
+import { ipRateLimit } from '@lib/ip-rate-limit'
+
+// run only on API routes
+export const config = {
+  matcher: '/api',
+}
+
+export async function middleware(req: NextRequest) {
+  const res = await ipRateLimit(req)
+  if (res.status !== 200) return res
```

Comment:
This won't work if you're rate limited because in that case the method sends a response that's not allowed by the new middleware.